pkg/runtime/runtime.go

```diff
@@ -3138,14 +3240,46 @@ func ShutdownSignal() chan os.Signal {
 	return stop
 }
 
+func (a *DaprRuntime) createHTTPEndpointsChannels() (map[string]channel.HTTPEndpointAppChannel, error) {
+	// Create dedicated app channels for known app endpoints
+	endpoints := a.compStore.ListHTTPEndpoints()
+
+	if len(endpoints) > 0 {
+		channels := make(map[string]channel.HTTPEndpointAppChannel, len(endpoints))
+
+		for _, e := range endpoints {
+			pipeline, err := a.buildAppHTTPPipeline()
+			if err != nil {
+				return nil, fmt.Errorf("failed to build app HTTP pipeline: %w", err)
+			}
+
+			conf, err := a.getHTTPEndpointAppChannel(pipeline, e)
+			if err != nil {
+				return nil, err
+			}
+
+			ch, err := httpChannel.CreateHTTPChannel(conf)
```

Comment:
Does it need closing to allow hot reloading?

Reply:
Yes in the future, but not for the hot-reloading being planned for the near future (1.12), which enables components. The work to enable hot-reloading for HTTP endpoints might need to include closing. It probably doesn't belong in this PR as there is no trigger to force the closing of a channel (nor is one being planned for 1.12).